client/src/components/Home/Media/SpotifyPlayer.jsx

```diff
@@ -51,36 +58,81 @@ const SpotifyPlayer = ({ token, trackUri }) => {
         console.error("Failed to perform playback", message);
       });
 
-      // Event listener for player state changes
       player.addListener("player_state_changed", (state) => {
         if (!state) return;
         setIsPaused(state.paused);
+        setTrackInfo({
+          name: state.track_window.current_track.name,
+          artist: state.track_window.current_track.artists
+            .map((artist) => artist.name)
+            .join(", "),
+          album: state.track_window.current_track.album.name,
+          image: state.track_window.current_track.album.images[0].url,
+        });
+        setProgress(state.position);
+        setDuration(state.duration);
+
+        if (!state.paused && !intervalId) {
+          const id = setInterval(() => {
+            setProgress((prevProgress) => prevProgress + 1000);
+          }, 1000);
+          setIntervalId(id);
+        } else if (state.paused && intervalId) {
+          clearInterval(intervalId);
+          setIntervalId(null);
+        }
       });
 
-      // Connect the player
```

Comment:
Is it possible we keep these comments which can help us understand the code better

Reply:
Yes thanks for the clarification